### PR TITLE
debug method

### PR DESCRIPTION
**Pull request by oddbookworm**
I saw #2540 and figured I'd make an attempt to implement it. I tested this on Manjaro and it worked properly for me, but I don't have a pygame dev environment set up in more than one place atm, so I can't test it in other places yet.

### CODE REVIEW

**Comment by oddbookworm** [comment]
In accordance with a conversation I had with @ankith26 in discord, I modified the PR to instead display the result of `platform.platform()` instead of the driver reported by sdl2

**Comment by Starbuck5** [comment]
Found a problem.

```py
>>> import os
>>> os.environ["PYGAME_HIDE_SUPPORT_PROMPT"] = "1"
>>> import pygame
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\charl\AppData\Local\Programs\Python\Python39\lib\site-packages\pygame\__init__.py", line 409, in <module>
    platform,
NameError: name 'platform' is not defined
>>>
```

**Comment by oddbookworm** [comment]
That's an easy fix

**Comment by oddbookworm** [comment]
@Starbuck5 should be fixed now

File: src_py/__init__.py
Comment (ankith26) [inline src_py/__init__.py:398]: This is not the architecture... it's a string with system OS info

**Review by ankith26 — COMMENTED** [review]
Thanks for the PR! 👍 

I support this change since it helps both issue reporters and issue triagers... but there is a minor downside that it makes the support prompt more verbose

**Comment by oddbookworm** [comment]
This PR is not dead, but is awaiting a pretty significant change according to a conversation in discord

**Comment by MyreMylar** [comment]
OK, I will hold off on my review until the significant change has appeared (cryptic :D)

**Comment by oddbookworm** [comment]
Waiting for #3379 to resolve. The future of this PR is something to the effect of:
A pygame.debug() method that gets everything relevant to the current setup. If display is initialized, it grabs the driver. If mixer is initialized, it grabs the SDL mixer info, etc.. and either prints to console or writes to a log file called pygame_debug.txt in the cwd (or the user can provide a filepath to save to instead?)

**Comment by MyreMylar** [comment]
@oddbookworm looks like this could be whipped into shape now that #3379 is merged.

**Review by MyreMylar — CHANGES_REQUESTED** [review]
Changes requested status: 

Update this to use #3379

**Comment by oddbookworm** [comment]
I'll work on this in the next few days or so

**Comment by oddbookworm** [comment]
Didn't mean to close the PR

**Comment by oddbookworm** [comment]
New summary:
A `pygame.debug` method that can optionally write to a file, but by default prints to the console. Retrieves all the important information about a particular user's setup

**Comment by oddbookworm** [comment]
If #3567 gets merged before this PR, then I don't need the special freetype logic and it'll behave like all of the others

**Comment by MyreMylar** [comment]
Example test program:

```
import pygame

pygame.init()
pygame.debug()
```

Example output:

```
Platform:		Windows-10-10.0.22000-SP0
System:			Windows
Processor:		AMD64 Family 23 Model 113 Stepping 0, AuthenticAMD
Architecture:		Bits: 64bit	Linkage: WindowsPE

pygame version:		2.1.4.dev1
python version:		3.9.0

SDL versions:		Linked: 2.24.0	Compiled: 2.24.0
SDL Mixer versions:	Linked: 2.6.2	Compiled: 2.6.2
SDL Font versions:	Linked: 2.20.1	Compiled: 2.20.1
SDL Image versions:	Linked: 2.0.5	Compiled: 2.0.5
Freetype versions:	Linked: 2.11.1	Compiled: 2.11.1
```

I like the output, should provide a useful block to copy and paste into issue reports. 

Should the function be called `debug_info()`, `print_debug_info()` or `print_system_info()` though instead? We have a few 'info' gathering functions (`display.Info()`, `display.get_wm_info()`, midi.get_device_info()) already that this would then fit among - though none of those also print straight to the console. 

I'm concerned that `debug()` is too general a name for what this function does which is essentially gather and print a bunch of hardware and software type & version information.

Getting names right is hard.

**Review by MyreMylar — APPROVED** [review]

**Review by ankith26 — COMMENTED** [review]
Probably also worth including `platform.python_implementation()` and probably `platform.version()` too in here

**Comment by ankith26** [comment]
> If https://github.com/pygame/pygame/pull/3567 gets merged before this PR, then I don't need the special freetype logic and it'll behave like all of the others

Cool, I think we should wait for that then before merging this one

**Comment by Starbuck5** [comment]
This code needs to be resilient against crashing in different possible pygame environments, including when different modules are unavailable or fail to import.

So I think this needs to handle the case of the sub modules being MissingModule, as well as printing the reason a module is missing (usually a trace back)

File: docs/reST/ref/pygame.rst
Comment (MyreMylar) [inline docs/reST/ref/pygame.rst:195]: does this documentation need updating to match the renamed function?

File: src_c/doc/pygame_doc.h
Comment (MyreMylar) [inline src_c/doc/pygame_doc.h:14]: again the name of the function looks to be the old name here.

File: docs/reST/ref/pygame.rst
Comment (oddbookworm) [reply docs/reST/ref/pygame.rst]: Just pushed a commit to fix this

File: src_c/doc/pygame_doc.h
Comment (oddbookworm) [reply src_c/doc/pygame_doc.h:14]: Just pushed a commit to fix this

**Review by MyreMylar — APPROVED** [review]
Current output looks like this on my machine:

```
Platform:		Windows-10-10.0.22000-SP0
System:			Windows
System Version:		10.0.22000
Processor:		AMD64 Family 23 Model 113 Stepping 0, AuthenticAMD
Architecture:		Bits: 64bit	Linkage: WindowsPE

Python:			CPython
pygame version:		2.1.4.dev1
python version:		3.11.1

SDL versions:		Linked: 2.24.0	Compiled: 2.24.0
SDL Mixer versions:	Linked: 2.6.2	Compiled: 2.6.2
SDL Font versions:	Linked: 2.20.1	Compiled: 2.20.1
SDL Image versions:	Linked: 2.0.5	Compiled: 2.0.5
Freetype versions:	Linked: 2.11.1	Compiled: 2.11.1
```

**Comment by oddbookworm** [comment]
> EDIT: I am on windows 11, confusingly the version number for windows 11 is still 10.0

I think this is a problem with windows 11 itself. The vast majority of programs report the OS as windows 10, so it has to be something in the OS

**Comment by MyreMylar** [comment]
Yes, sorry my comment was a bit confusing. The debug printout is correct,
it is just Microsoft being confusing with their versioning.

**Review by MyreMylar — APPROVED** [review]

**Pull request closed without merging**
